dataset/src/main/scala/frameless/TypedColumn.scala

```diff
@@ -451,11 +451,16 @@ object TypedColumn {
   object Exists {
     def apply[T, V](column: Witness)(implicit e: Exists[T, column.T, V]): Exists[T, column.T, V] = e
 
-    implicit def deriveRecord[T, H <: HList, K, V]
+    implicit def deriveProduct[T, H <: HList, K, V]
       (implicit
         i0: LabelledGeneric.Aux[T, H],
```

Comment:
Kind of funny that you cannot go from a Record to a Record using generic deriviation.

Reply:
Yeah. There's no identity typeclass instance of `LabelledGeneric`.

Reply:
It's intentional that there isn't a `LabelledGeneric` for record types; it would cause divergence issues for anyone using generic derivation.

Reply:
@jeremyrsmith yeah ... now I see it ... makes sense. At the same time, you would expect something that works for a case class that gets converted implicitly to LabelledGeneric to work with the LabelledGeneric passed explicitly ...